spec/compiler/parser/parser_spec.cr

```diff
@@ -32,6 +32,15 @@ private def assert_end_location(source, line_number = 1, column_number = source.
   end
 end
 
+private def parser_warnings(code)
+  warnings = WarningCollection.new
+  parser = Parser.new(code, warnings: warnings)
+  parser.filename = "test.cr"
+  node = parser.parse
+
+  warnings
+end
+
```

Comment:
There are existing specs for parser warnings in `spec/compiler/parser/warnings_spec.cr` already; consider moving the tests in this file over there or vice versa